awx/main/access.py

```diff
@@ -2769,5 +2779,75 @@ def can_delete(self, obj):
         return False
 
 
+class WorkflowApprovalAccess(BaseAccess):
+    '''
+    A user can create an approval template if they are a superuser, an org admin
+    of the org connected to the workflow, or if they are assigned as admins to
+    the workflow.
+
+    A user can approve a workflow when they are:
+    - a superuser
+    - a workflow admin
+    - an organization admin
+    - any user who has explicitly been assigned the "approver" role
+    '''
+
+    model = WorkflowApproval
+    prefetch_related = ('created_by', 'modified_by',)
+
+    def can_use(self, obj):
+        return True
+
+    def can_start(self, obj, validate_license=True):
+        return True
+
+    def filtered_queryset(self):
+        return self.model.objects.filter(
+            unified_job_node__workflow_job__unified_job_template__in=WorkflowJobTemplate.accessible_pk_qs(
```

Comment:
@AlanCoding what do you think about this jump from `WorkflowJob` -> `UnifiedJobTemplate`.  Do you think this is appropriate, or should we change this slightly (e.g., what happens if the `WorkflowApprovalTemplate` gets deleted)?

Reply:
I think this is fine.

> what happens if the WorkflowApprovalTemplate gets deleted

Then only superusers will be able to view.

If both this and https://github.com/ansible/awx/pull/4338 are merged, then we can rewrite this to allow org admins access as well.